.github/workflows/lint.yaml

```diff
@@ -51,20 +37,8 @@ jobs:
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
         continue-on-error: true
 
-      - name: restore cargo cache
-        uses: actions/cache@v3
-        with:
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock', 'rust-toolchain.toml') }}
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-            .cargo/.package-cache/
-
-      - name: Install dependencies
-        run: nix develop -c make -j deps-ci
+      - name: Build bindings
+        run: nix develop -c make build-solidity-types
```

Comment:
The bindings build should be done as a cargo build script, this is way overdue and would simplify our life a lot: https://doc.rust-lang.org/cargo/reference/build-scripts.html

Reply:
https://doc.rust-lang.org/cargo/reference/build-script-examples.html#code-generation

Reply:
Gonna do this in a separate mini-PR